rpcs3/Emu/Cell/lv2/sys_spu.cpp

```diff
@@ -1219,29 +1219,34 @@ error_code sys_spu_thread_bind_queue(ppu_thread& ppu, u32 id, u32 spuq, u32 spuq
 
 	std::lock_guard lock(thread->group->mutex);
 
+	decltype(std::data(thread->spuq)) q{};
+
 	for (auto& v : thread->spuq)
 	{
-		if (auto q = v.second.lock())
+		// Check if the entry is assigned at all
+		if (const decltype(v.second) test{};
+			!v.second.owner_before(test) && !test.owner_before(v.second))
 		{
-			if (v.first == spuq_num || q == queue)
-			{
-				return CELL_EBUSY;
-			}
+			// Use the empty entry later
+			q = &v;
+			continue;
 		}
-	}
 
-	for (auto& v : thread->spuq)
-	{
-		if (v.second.expired())
+		if (v.first == spuq_num || v.second.owner_before(queue))
```

Comment:
```suggestion
		if (v.first == spuq_num || (!v.second.owner_before(queue) && !queue.owner_before(v.second)))
```

Reply:
Shouldn't the condition be inverted? it gives wrong results in testcase.